packages/components/src/internal/components/assay/AssayImportPanels.tsx

```diff
@@ -638,31 +606,39 @@ class AssayImportPanelsBody extends Component<Props, State> {
                 return data.dataRows.size * 10;
             }
         }
-    }
+    };
 
     onCancelRename = (): void => {
-        this.setState(() => ({
-            showRenameModal: false,
-            duplicateFileResponse: undefined,
-            importAgain: undefined,
-        }));
+        this.setState(() => ({ showRenameModal: false, duplicateFileResponse: undefined, importAgain: undefined }));
     };
 
     onRenameConfirm = (): void => {
-        const { showRenameModal, importAgain } = this.state;
+        const { importAgain } = this.state;
+        this.setState(
+            () => ({ showRenameModal: false, duplicateFileResponse: undefined, importAgain: undefined }),
+            () => this.onFinish(importAgain)
+        );
+    };
 
-        if (showRenameModal) {
-            this.setState(
-                () => ({
-                    showRenameModal: false,
-                    duplicateFileResponse: undefined,
-                    importAgain: undefined,
-                }),
-                () => {
-                    this.onFinish(importAgain);
-                }
-            );
-        }
+    onGridChange = (
```

Comment:
Yes, but I think we can find a way to make it easier to manage. For example I could see us making something like `useEditableGridState(queryModel)` which returns an `EditorModel` based on the passed in QM, as well as the basic actions for updating the model. We basically have to have it this way because many consumers rely on using the EditorModel (e.g. to grab the data then POST it to the server).